src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -74,24 +72,24 @@ public BrotliStream(Stream baseStream, CompressionMode mode, bool leaveOpen = fa
             _mode = mode;
             _stream = baseStream;
             _leaveOpen = leaveOpen;
+            _state = new Brotli.State();
             if (_mode == CompressionMode.Compress)
             {
-                _encoder = new Encoder();
-                _encoder.SetQuality();
-                _encoder.SetWindow();
+                _state.InitializeEncoder();
```

Comment:
In the stream case, I would say that an exception bubbling up is probably a good thing, unless you plan on wrapping and re-throwing.